avocado/utils/data_structures.py

```diff
@@ -253,6 +260,20 @@ def time_to_seconds(time):
     return seconds
 
 
+def size_to_bytes(size):
+    """
+    Convert size with suffix to bytes
+    :param size: String optionally with suffix (i.e. '1M')
+    """
+    norm_size = size.strip().lower()
+    last = norm_size[-1]
+    if last.isdigit():
+        return int(norm_size)
+    elif last in SIZE_MULTIPLIERS:
+        return int(norm_size[:-1]) * SIZE_MULTIPLIERS[last]
+    raise ValueError("Unable to convert '%s'" % size)
+
+
```

Comment:
I don't see a reason for this to exist... There are no such thing as 'multiple conversions' in `DataSize()`. The conversions happen on-demand, when you call the corresponding attribute. `DataSize('1M').b` has no additional overhead when compared to this `size_to_bytes()`.

Reply:
Maybe all you need is a wrapper function to `DataSize` to fit in the ArgumentParser `type`?

Reply:
Actually I wanted to do the opposite, but as the conversion is ad-hock we can't really share the code. The thing is that DataSize is a generic, fat object, that consumes lot of memory (ok, most of it is there just once) and even more cycles.

Take those two examples:

```python
# a.py
from avocado.utils.data_structures import size_to_bytes
for i in xrange(10000000):
    size_to_bytes("%sm" % i)
```

```python
# b.py
from avocado.utils.data_structures import DataSize
for i in xrange(10000000):
    DataSize("%sm" % i).b
```

I'm getting those times:

```
/usr/bin/time python /tmp/a.py
7.60user 0.02system 0:07.64elapsed 99%CPU (0avgtext+0avgdata 22996maxresident)k
0inputs+0outputs (0major+4456minor)pagefaults 0swaps

/usr/bin/time python /tmp/b.py 
32.32user 0.01system 0:32.40elapsed 99%CPU (0avgtext+0avgdata 22784maxresident)k
0inputs+0outputs (0major+4440minor)pagefaults 0swaps
```

(that's why I never really the original DataSize, in most cases I'm interested in single conversion and not multiple ones to benefit from the object...)

Reply:
When you call `DataSize("%sm" % i).b`, there's only a single conversion, not multiple ones.
Ok, your method is faster (probably due to the parameter/unit treatment difference), but the outcome is still an `int`, just like the outcome from `DataSize("%sm" % i).b`, and that (usually) means a 24bytes object.
As for the assumption that the 'DataSize' object is fat, its instance created as part of the call to `DataSize().b` will not last long, since no one will reference that object and it will be garbage collected soon enough.
Maybe you could just use the knowledge you have in making the parameter/unit treatment faster to improve the `DataSize` object instead?

Reply:
I spent some time looking into the DataSize performance and the real reason it's slower than the standalone function you created is the use of the `__getattr__` to dynamically create the class attributes. As `__getattr__` is a fallback method, it's only called after the attribute is looked-up & not found, as part of the exception handling. Indeed an expensive way to create attributes.
I will try to figure out a way to have the best of both worlds: speed and dynamically created attributes. A great engineering exercise :)

Reply:
The aim of both is different. We already have `time_to_seconds` so I believe it makes sense to have simple implementation of `size_to_bytes` while still keeping the DataSize for advanced conversions (no matter how optimized it is you still divide the ranges (n/1) to get the conversion compare the base conversion)

Reply:
So, if no matter how optimized it is, you still want to use a function with redundant functionality, then my work is done here.

Reply:
@ldoktor can you describe how the *aim* of `size_to_bytes` and `DataSize` is different? Not getting in the implementation details, they seem to *aim* or *fulfill* the same use case.

If they're not different, it's much worse to have two implementations than have one that "performs worse" than the other.  This has to do with API design, and how users would be confused by multiple APIs to do the same thing.

Reply:
@clebergnu: The aim of `size_to_bytes` is to be the analogy of `time_to_seconds`. Easy to use, single-purpose extremely fast function to get the usual default size unit. The aim of DataSize is to be able to calculate with size object of any unit and combine it with size objects of various units (or just do conversion). It's powerful, does not lose the main unit, quite fast when various units are used, but it can't be as simple and quick as simple 2bytes implementation. I don't think having both would be confusing, DataSize supports several units and `size_to_bytes` is yet another simple method to get bytes from anything (like `time_to_seconds`). Whether the implementation is via `DataSize` or separate is but a minor detail.

@apahim: I don't say it's impossible to optimize it to the same extend as the simple function, but it's very unlikely.

Reply:
@ldoktor when I said "aim", I meant the use case, as in which input should it take, and what output should it produce.  It's clear that both `size_to_bytes` and `DataSize` (can) fulfill the exact same use case, right?

Also, when it comes to developer usability, both can be used with the same level of effort, right? I mean:

```
value = DataSize('1m').b
```

is not harder or easier to use then:

```
value = size_to_bytes('1m')
```